infra/src/Pos/Infra/Network/DnsDomains.hs

```diff
@@ -38,7 +62,111 @@ data NodeAddr a =
     -- If domain unspecified, use the node's name
     -- If port unspecified, use the default.
   | NodeAddrDNS a (Maybe Word16)
-  deriving (Show)
+  deriving (Eq, Generic, Show)
+
+instance ToJSON (NodeAddr (Maybe DNS.Domain)) where
+    toJSON =
+        A.object . \case
+            (NodeAddrExact ip mWord) ->
+                case (ip, mWord) of
+                    (IPv4 (IP4 ip4), Just port) ->
+                        [ "addr" .= (A.String . show $ IPv4 (IP4 ip4))
+                        , "port" .= (toJSON port)
+                        ]
+                    (IPv4 (IP4 ip4), Nothing)   ->
+                        [ "addr" .= (A.String . show $ IPv4 (IP4 ip4))
+                        , "port" .= (toJSON (Prelude.read "3000" :: Word16))
+                        ]
+                    (IPv6 (IP6 ip6), Just port) ->
+                        [ "addr" .= (A.String . show $ IPv6 (IP6 ip6))
+                        , "port" .= (toJSON port)
+                        ]
+                    (IPv6 (IP6 ip6), Nothing)   ->
+                        [ "addr" .= (A.String . show $ IPv6 (IP6 ip6))
+                        , "port" .= (toJSON (Prelude.read "3000" :: Word16))
+                        ]
+            (NodeAddrDNS nHost mWord) ->
+                case (nHost, mWord) of
+                    (Just host, Just port) ->
+                        [ "host" .= (toJSON $ decodeUtf8 @Text host)
+                        , "port" .= (toJSON port)
+                        ]
+                    (Just host, Nothing)   ->
+                        [ "host" .= (toJSON $ decodeUtf8 @Text host)
+                        , "port" .= (toJSON (Prelude.read "3000" :: Word16))
+                        ]
+                    (Nothing, _)      -> error "Please enter hostname"
```

Comment:
@intricate I think that's more clear, since the viewer of the log doesn't have some box to "enter [a] hostname" into, and it's a change in either source code or a config file.